src/arguments.c

```diff
@@ -52,7 +52,7 @@
 #define BL_EXTRA        2   /* Bootloader at top in separate memory area */
 #define BL_SPECIFIC     3   /* Any value greater than this is a specific start address */
 
-extern int debug;       /* defined in main.c */
+extern int debug;       /* defined in libdfu.c */
```

Comment:
Would it make more sense to define this in libdfu.h rather than an extern?

Reply:
I was trying to keep the public interface as plain as possible. But maybe `debug` should be exposed? Or should we fix its thread safety?

Reply:
I would say add it to args and maybe use an accessor function defined in libdfu.c to read its value in those places that don't receive the args structure directly?

Reply:
That works for me. More explicitly fixes the multithreading issue.